s3/src/test/scala/com/meltwater/fawn/s3/S3PayloadsSpec.scala

```diff
@@ -0,0 +1,281 @@
+      | <DisplayName>accountname</DisplayName>
+      | <ID>accountid</ID>
+      | </Owner>
+      |</ListAllMyBucketsResult>""".stripMargin
+
+  test("decode a list all my buckets response") {
+    EntityDecoder[IO, Elem]
+      .decode(
+        Response[IO]()
+          .withEntity(listBucketsExample)
+          .withHeaders(Header("content-type", "text/xml")),
+        false
+      )
+      .value
+      .flatMap { r =>
+        IO {
+          r.map { elem =>
+            ListBucketsResponse.xmlDecoder.read(elem)
+          }.isRight
+        }.assertEquals(true)
+      }
+  }
+
+  val listObjectsExample: String =
+    """<?xml version="1.0" encoding="UTF-8"?>
+      |<ListBucketResponse>
+      | <Name>bucket</Name>
+      | <KeyCount>205</KeyCount>
+      | <MaxKeys>1000</MaxKeys>
+      | <IsTruncated>false</IsTruncated>
+      | <Contents>
+      |  <Key>my-image.jpg</Key>
+      |  <LastModified>2009-10-12T17:50:30.000Z</LastModified>
+      |  <ETag>"fba9dede5f27731c9771645a39863328"</ETag>
+      |  <Size>434234</Size>
+      |  <StorageClass>STANDARD</StorageClass>
+      | </Contents>
+      |</ListBucketResponse>""".stripMargin
+
+  test("decode a list all objects within a bucket response") {
+    EntityDecoder[IO, Elem]
+      .decode(
+        Response[IO]()
+          .withEntity(listObjectsExample)
+          .withHeaders(Header("content-type", "text/xml")),
+        false
+      )
+      .value
+      .flatMap { r =>
+        IO {
+          r.map { elem =>
+            ListObjectsResponse.xmlDecoder.read(elem)
+          }.isRight
+        }.assertEquals(true)
+      }
+  }
+
+  val copyObjectExample: String =
+    """<?xml version="1.0" encoding="UTF-8"?>
+      |<CopyObjectResult>
+      |  <ETag>"9b2cf535f27731c974343645a3985328"</ETag>
+      |  <LastModified>2009-10-28T22:32:00</LastModified>
+      |</CopyObjectResult>""".stripMargin
+
+  test("decode a copy object response") {
+    EntityDecoder[IO, Elem]
+      .decode(
+        Response[IO]()
+          .withEntity(copyObjectExample)
+          .withHeaders(Header("content-type", "text/xml")),
+        false
+      )
+      .value
+      .flatMap { r =>
+        IO {
+          r.map { elem =>
+            CopyObjectResponse.xmlDecoder.read(elem)
+          }.isRight
+        }.assertEquals(true)
+      }
+  }
+
+  val createMultipartUploadExample: String =
+    """<?xml version="1.0" encoding="UTF-8"?>
+      |<InitiateMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+      |   <Bucket>example-bucket</Bucket>
+      |   <Key>example-object</Key>
+      |   <UploadId>VXBsb2FkIElEIGZvciA2aWWpbmcncyBteS1tb3ZpZS5tMnRzIHVwbG9hZA</UploadId>
+      |</InitiateMultipartUploadResult>
+      |""".stripMargin
+
+  test("decode a create multipart upload response") {
+    EntityDecoder[IO, Elem]
+      .decode(
+        Response[IO]()
+          .withEntity(createMultipartUploadExample)
+          .withHeaders(Header("content-type", "text/xml")),
```

Comment:
prefer `.transformHeaders(_.put(Header(...)))` so it doesn't overwrite existing headers